src/conductor/optimize-imports.ts

```diff
@@ -0,0 +1,71 @@
+import ts from 'typescript';
+import chalk from 'chalk';
+import { getConfig } from '../config';
+import simpleGit, { SimpleGit } from 'simple-git';
+import { readFileSync, writeFileSync } from 'fs';
+import { collectImportNodes } from './collect-import-nodes';
+import { getImportStatementMap } from './get-import-statement-map';
+import { categorizeImportLiterals } from './categorize-imports';
+import { sortImportCategories } from './sort-import-categories';
+import { formatImportStatements } from './format-import-statements';
+
+const git: SimpleGit = simpleGit();
+
+export const actions = {
+  none: 'none',
+  skipped: 'skipped',
+  reordered: 'reordered',
+};
+
+export async function optimizeImports(filePath: string): Promise<string> {
+  if (!/\.tsx?$/.test(filePath)) {
+    return actions.none;
+  }
+
+  const fileContent = readFileSync(filePath);
```

Comment:
could we do this earlier? - like we shouldn't optimize the imports at all if the file is ignored.

Reply:
we are not optimizing if he is ignored (we return with `actions.skipped`), I'm not sure what you mean.